2-mandatory/2-retrying-random-numbers.js

```diff
@@ -8,11 +8,23 @@ function generateRandomNumber() {
     console.log("Generating number...");
     return Math.round(Math.random() * 100);
 }
+// let randomNum = generateRandomNumber;
 
 function getRandomNumberGreaterThan50() {
     // TODO - implement using a do-while loop
-}
+    let i = 100;
+    // let randomNumber = generateRandomNumber;
+    do {
+        let random = generateRandomNumber();
+        if (random > 50) {
+            return random;
+        }
 
+        i--;
+    } while (i > 50);
```

Comment:
This will exit after 50 loops, so it might not always find a number over 50. But I also understand why you want it to exit so that you don't end up with an endless loop if your logic is wrong.